crates/core/src/operations/merge/filter.rs

```diff
@@ -453,14 +453,7 @@ mod tests {
         .build()
         .unwrap();
 
-        let join_predicate = col(Column {
```

Comment:
the key rationale for this change is that `Column` has a new field in DataFUsion 46, so if it is constructed explicitly like this I need to change all the callsites

If I switch to using `Column::new()` then nothing needs to be changed as part of the upgrade (as Column::new handles initializing the newly added field)